### PR TITLE
Add markdown file for using the Google sign-in button

### PR DESCRIPTION
**Pull request by mdmathias**

### CODE REVIEW

**Review by Alex-4-Git — APPROVED** [review]

File: UsingGoogleSignInButton.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
+# Using the Google Sign In Button
+
+Depending upon your platform, GoogleSignIn-iOS provides a few options to use
+when creating a sign-in button.
+
+## SwiftUI
```
Comment (petea) [inline UsingGoogleSignInButton.md:6]: What if we went with a UI framework focused structure?

## SwiftUI (iOS and macOS)
## UIKit (iOS)
## AppKit (macOS)

File: UsingGoogleSignInButton.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+If you are not using SwiftUI to build your user interfaces, you can either
+create `GIDSignInButton` programmatically, or in a Xib/Storyboard. 
+If you are writing programmatic UI code, it will look something like this:
+
+`GIDSignInButton *button = [[GIDSignInButton alloc] initWithFrame: CGRectMake(<YOUR_RECT>)];`
+
+## macOS
+
+Given that `GIDSignInButton` is implemented as a subclass of `UIControl`, it
```
Comment (petea) [inline UsingGoogleSignInButton.md:35]: Might want to briefly address `NSHostingView` and the problem it solves here.

File: UsingGoogleSignInButton.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
+# Using the Google Sign In Button
```
Comment (petea) [inline UsingGoogleSignInButton.md:1]: How should we link this content to README.md?  Alternatively, would it be better to merge it into README.md?

File: UsingGoogleSignInButton.md [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+Depending upon your platform, GoogleSignIn-iOS provides a few options to use
+when creating a sign-in button.
+
+## SwiftUI
+
+Creating a Google sign-in button in SwiftUI can be as simple as this:
+
+```
+GoogleSignInButton {
+  // Whatever action you'd like to take on tap
```
Comment (petea) [inline UsingGoogleSignInButton.md:12]: Should we provide some pseudo code for extra context?

File: UsingGoogleSignInButton.md (hunk 1 above)
Comment (mdmathias) [reply UsingGoogleSignInButton.md:6]: Sure thing.

File: UsingGoogleSignInButton.md (hunk 2 above)
Comment (mdmathias) [reply UsingGoogleSignInButton.md:35]: Good suggestion.

File: UsingGoogleSignInButton.md (hunk 3 above)
Comment (mdmathias) [reply UsingGoogleSignInButton.md:1]: I will add it to the `README`.

File: UsingGoogleSignInButton.md (hunk 4 above)
Comment (mdmathias) [reply UsingGoogleSignInButton.md:12]: I've added some surrounding code view-related code. Is that what you mean? IMHO, I think it's a little strange to have it for the SwiftUI section and not the others. Adding the same to the others feels like we're tutorializing how to add views to UIs in AppKit, UIKit, and SwiftUI, which isn't really the goal here. That's why I didn't have that context before.

Let me know what you think.

Comment (petea) [reply UsingGoogleSignInButton.md:12]: I'm suggesting that we show how you would kick off a sign-in flow on tap.

File: UsingGoogleSignInButton.md [hunk 5]
```diff
@@ -0,0 +1,46 @@
+# Using the Google Sign In Button
```
Comment (petea) [inline UsingGoogleSignInButton.md:1]: Should we drop this file now?

File: UsingGoogleSignInButton.md (hunk 4 above)
Comment (mdmathias) [reply UsingGoogleSignInButton.md:12]: From offline conversation, I've added a call to `GIDSignIn.sharedInstance.signIn` in the action for the `GoogleSignInButton`.

File: UsingGoogleSignInButton.md (hunk 5 above)
Comment (mdmathias) [reply UsingGoogleSignInButton.md]: Yep

File: README.md [hunk 6]
```diff
@@ -63,3 +63,58 @@ Google Sign-In also supports iOS apps that are built for macOS via
 [Mac Catalyst](https://developer.apple.com/mac-catalyst/).  In order for your Mac Catalyst app
 to store credientials via the Keychain on macOS, you will need to
 [sign your app](https://developer.apple.com/support/code-signing/).
+
+# Using the Google Sign In Button
```
Comment (petea) [inline README.md:67]: Since the README title is H1, should we make this H2 and the sections below H3?

Comment (petea) [inline README.md:67]: "Sign-In"

File: README.md [hunk 7]
```diff
@@ -63,3 +63,58 @@ Google Sign-In also supports iOS apps that are built for macOS via
...
+[This convenience initializer](GoogleSignInSwift/Sources/GoogleSignInButton.swift#L56) 
+provides parameters that you can use to set these values as needed.
+
+### UIKit (iOS)
+
+If you are not using SwiftUI to build your user interfaces, you can either
+create `GIDSignInButton` programmatically, or in a Xib/Storyboard. 
+If you are writing programmatic UI code, it will look something like this:
+
+`GIDSignInButton *button = [[GIDSignInButton alloc] initWithFrame: CGRectMake(<YOUR_RECT>)];`
```
Comment (petea) [inline README.md:101]: Should we provide this code sample in Swift to match the other two?

File: README.md [hunk 8]
```diff
@@ -63,3 +63,58 @@ Google Sign-In also supports iOS apps that are built for macOS via
 [Mac Catalyst](https://developer.apple.com/mac-catalyst/).  In order for your Mac Catalyst app
 to store credientials via the Keychain on macOS, you will need to
 [sign your app](https://developer.apple.com/support/code-signing/).
+
+## Using the Google Sign-In Button
+
+Depending upon your platform, GoogleSignIn-iOS provides a few options to use
```
Comment (petea) [inline README.md:69]: "There are several paths to adding a 'Sign in with Google' button to your app, which path you choose will depend on your UI framework and target platform."

File: README.md [hunk 9]
```diff
@@ -63,3 +63,58 @@ Google Sign-In also supports iOS apps that are built for macOS via
...
+when creating a sign-in button.
+
+### SwiftUI (iOS and macOS)
+
+Creating a Google sign-in button in SwiftUI can be as simple as this:
```
Comment (petea) [inline README.md:74]: Google sign-in button -> 'Sign in with Google' button

File: Samples/Swift/DaysUntilBirthday/Shared/Services/GoogleSignInAuthenticator.swift [hunk 10]
```diff
@@ -47,28 +47,37 @@ final class GoogleSignInAuthenticator: ObservableObject {
       return
     }
 
-    GIDSignIn.sharedInstance.signIn(with: configuration,
-                                    presenting: rootViewController) { user, error in
-      guard let user = user else {
+    Task {
```
[the remark this answers is not in the document]
Comment (mdmathias) [reply Samples/Swift/DaysUntilBirthday/Shared/Services/GoogleSignInAuthenticator.swift:50]: Argh. Good catch. I did not.

File: README.md (hunk 7 above)
Comment (mdmathias) [reply README.md]: Sure thing

File: Samples/Swift/DaysUntilBirthday/Shared/Services/GoogleSignInAuthenticator.swift (hunk 10 above)
[the remark this answers is not in the document]
Comment (petea) [reply Samples/Swift/DaysUntilBirthday/Shared/Services/GoogleSignInAuthenticator.swift]: I'm still seeing this in the PR.

[the remark this answers is not in the document]
Comment (mdmathias) [reply Samples/Swift/DaysUntilBirthday/Shared/Services/GoogleSignInAuthenticator.swift]: I've reverted the changes. Should be gone from this PR.

**Review by petea — APPROVED** [review]

**Pull request merged**